ls1intum/ArTEMiS
# Extend tutor dashboard with modeling exercises and add example solution

### Checklist

### Motivation and Context
With growing numbers of students that use ArTEMiS an increasing number of tutors is necessary to assess the students' submissions. All these tutors have different levels of experience and knowledge. In order to increase the assessment quality and to decrease the number of conflicting assessments for modeling exercises, we want to improve the training of tutors by a tutor dashboard including assessment tutorials.

### Description
We extend the existing tutor dashboard to also support modeling exercises. In the dashboard tutors have to read the grading instructions and review or assess example submissions created by the instructor. After completing these steps, tutors are able to assess real student submission.
We additionally extend modeling exercises by an example solution and a corresponding example solution explanation which can be added to the exercise via the edit exercise dialog. It is displayed in the instructions section in the assessment editor and should help tutors with modeling assessments.

### Steps for Testing
1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Go to Exercises
4. Create a new modeling exercise
5. Edit the previously created exercise
6. Add grading instructions
7. Add example solution and explanation
8. Add example submissions and assessments, at least one with and one without "Used for Tutorial" flag
9. Navigate to Course Administration
10. Go to Tutor Dashboard
11. Open Exercise Dashboard for previously created Exercise
12. Read and confirm grading instructions
13. Review example submission(s) not "Used for Tutorial"
14. Assess example submission(s) "Used for Tutorial"
15. See student submission appear in the dashboard after completing the steps above
16. Assess student submission

Note, that the example solution and explanation is displayed in the instructions section on the right in the review and assess example submission views and in the assessment editor for student submissions.

### Screenshots
Example Solution in Edit Exercise Dialog
![1 Example Solution in Edit Exercise Dialog](https://user-images.githubusercontent.com/5919138/56082548-5c2b1a00-5e1a-11e9-8a28-091107670c51.JPG)

Example Solution Editor
![2 Example Solution Editor](https://user-images.githubusercontent.com/5919138/56082552-75cc6180-5e1a-11e9-881d-e3e1807e1bd3.JPG)

Read Grading Instructions
![3  Read Grading Instructions](https://user-images.githubusercontent.com/5919138/56082554-7e249c80-5e1a-11e9-8ad2-646b1d10b707.JPG)

Review Example Submission
![5  Review Example Submission](https://user-images.githubusercontent.com/5919138/56082559-8e3c7c00-5e1a-11e9-8326-0cfba6955f54.JPG)

Assessment of Example Submission
![6  Assess Example Submission](https://user-images.githubusercontent.com/5919138/56082561-95fc2080-5e1a-11e9-8924-cec87dac1d67.JPG)

Student Submission Overview
![7  Student Submission Overview](https://user-images.githubusercontent.com/5919138/56082562-9d232e80-5e1a-11e9-870e-3029513073f9.JPG)

Assessment of Student Submission
![8  Assessment of Student Submission](https://user-images.githubusercontent.com/5919138/56082564-a4e2d300-5e1a-11e9-8a47-4abd7439f131.JPG)
